public/samples/gmp-senderreceiver.sol

```diff
@@ -26,10 +26,11 @@ contract SenderReceiver is AxelarExecutable {
             msg.sender
         );
 
-        gateway.callContract(destinationChain,destinationAddress,payload);
```

Comment:
Can you explain why we function call the gateway rather than referring to it now? I don't fully get it.

Also we should pin the imports of dependencies to a version at the top.

Reply:
It's because the upgraded executable contract does not include a `gateway()` that returns the address of the gateway directly. See [here](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/b5d0b7bdda0437fce983daffb776669437b809d0/contracts/executable/AxelarExecutable.sol#L71). The same is true for `gatewayWithToken()`.